c10/xpu/XPUMacros.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#ifndef C10_USING_CUSTOM_GENERATED_MACROS
+
+// Here is used to define the macro C10_XPU_BUILD_SHARED_LIBS
+#include <c10/xpu/impl/xpu_cmake_macros.h>
+
+#endif
+
+// See c10/macros/Export.h for a detailed explanation of what the function
+// of these macros are.  We need one set of macros for every separate library
+// we build.
+
+#ifdef _WIN32
+#if defined(C10_XPU_BUILD_SHARED_LIBS)
+#define C10_XPU_EXPORT __declspec(dllexport)
+#define C10_XPU_IMPORT __declspec(dllimport)
+#else
+#define C10_XPU_EXPORT
+#define C10_XPU_IMPORT
+#endif
```

Comment:
Why this #ifdef is needed if XPU is not available on Windows yet, is it?

Reply:
Let me remove it.